Corrected typings for openQuickCreate, attribute.setIsValid methods and AddNotificationObject.

sergeytunnik opened the pull request:
Closes #179, #183, #184.

sergeytunnik committed:
Merge pull request #3 from delegateas/gh_master

upstream

sergeytunnik committed:
Corrected typings for openQuickCreate, attribute.setIsValid methods and AddNotificationObject.

magesoe commented on src/XrmDefinitelyTyped/Resources/Extensions/xrm_ext_9-.d.ts:
```diff
@@ -1227,7 +1227,7 @@ declare namespace Xrm {
 
     interface AddNotificationObject {
-        actions?: actionsObject | null;
+        actions?: actionsObject[] | null;
```
Is the null needed? The argument is already optional. The same applies to the actionsObject typing. I know that is how it was, but we might aswell fix it now :)

sergeytunnik replied:
Checked and there is no need to specify nulls for optionals in typescript. Also done smoke checking that actual addNotification method doesn't throw when actions or messages are not specified.

sergeytunnik committed:
Optional attributes are already nullable.

TomMalow committed:
Split the different implementation of openQuickCreate.
Old implementation is in version 7.1 up to, but not inlcuding 8.2.
New implementation is in version 8.2 up to 9 where it becomes depricated.

TomMalow commented:
I believe that the move from callbacks functions to .then() started in 8.2. But I highly doubt that .then() works version below 8.2. The related issue #179 also only mention 8.2.
Until we know if it is also an issue in older version, then I made the choice of splitting the old definition and new definition out to target different version of CRM.

Old definition: Added in 7, removed in 8.2
New definition: Added in 8.2, deprecated in 9.
